app/src/main/java/com/saotome/imed_app/ui/fragments/DrugListAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package com.saotome.imed_app.ui
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewbinding.ViewBinding
+import com.saotome.imed_app.databinding.ItemDrugBinding
+import com.saotome.imed_app.model.Drug
+
+class DrugListAdapter : ListAdapter <Drug, DrugListAdapter.DrugViewHolder>(DiffCallback()) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DrugViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding = ItemDrugBinding.inflate(inflater, parent, false)
+        return DrugViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: DrugViewHolder, position: Int) {
+        holder.bind(getItem(position))
+    }
+
+    class DrugViewHolder(
+        private val binding: ItemDrugBinding
+    ) : RecyclerView.ViewHolder(binding.root) {
+        fun bind(item: Drug) {
+            binding.tvMedicationName.text = item.name
+            binding.tvDosage.text = item.dosage
+            binding.tvFrequency.text = item.frequency
+            binding.tvStartTime.text = item.startTime
+        }
+    }
+}
+
+class DiffCallback : DiffUtil.ItemCallback<Drug> () {
+    override fun areItemsTheSame(oldItem: Drug, newItem: Drug): Boolean = oldItem == newItem
+
+    override fun areContentsTheSame(oldItem: Drug, newItem: Drug): Boolean = oldItem.id == newItem.id
+
```

Comment:
Remove this line

Reply:
Done
ac033076add9121739da75bacc1053902964c6a8